Sources/Property.swift

```diff
@@ -5,11 +5,8 @@ import Glibc
 #endif
 import enum Result.NoError
 
-/// Represents a property that allows observation of its changes.
-///
-/// Only classes can conform to this protocol, because having a signal
-/// for changes over time implies the origin must have a unique identity.
-public protocol PropertyProtocol: class, BindingSource {
+
+public protocol BasePropertyProtocol: class, BindingSource {
```

Comment:
So I thought using some intermediate 'base' protocol would help reduce the difference between swift 4.0 and swift 4.1.  But it introduces a new public Protocol name.  Not sure if this adds complexity or reduces it in the whole scope of things.

Reply:
I think this makes sense 👍

Reply:
I think this makes sense, but I would probably name this `_BasePropertyProtocol` to indicate that it's not a type that's meant to be used outside the codebase (just like Swift's stdlib does when they hit language limitations), especially considering that we'll be removing this workaround when we drop `Swift 4.0.x` support.